### DISCUSSION TITLE
Global Styles

### DISCUSSION THREAD

**Discussion by lmtr0** [lit/lit#2658] [Q&A] [answered]
Hello there, I would like to use global styles with a lit component. I tried removing the shadow DOM, but it worked partially. What's the correct way of doing this?

In stencil, the end result of:
```js
@Component({
  tag: 'my-component',
  shadow: false,
})
export class MyComponent {
  @Prop() count: number = 0;

  private _onClick() {
    this.count += 1;
  }

  render() {
    return <button class="btn btn-primary" onClick={() => this._onClick()}>
      <slot />
      {this.count}
    </button>;
  }
}
```
being used like this:
```html
<my-component>
      <span> Count in</span>
</my-component>
```
is
![image](https://user-images.githubusercontent.com/57605930/159590375-d62abc45-e04e-438e-b89b-fe2bd9d53352.png)

However, with lit, this:
```ts
@customElement('my-element')
export class MyElement extends LitElement {
  @property({ type: Number })
  count = 0

  protected createRenderRoot(): Element | ShadowRoot {
    return this.attachShadow({slotAssignment: 'named', mode:"open", delegatesFocus: false});
  }

  render() {
    return html`
      <button class="btn btn-primary" @click=${() => this._onClick()}>
        Count: ${this.count}
        <slot></slot>
      </button>
    `
  }

  private _onClick() {
    // console.log("the click was processed")
    this.count++
  }
}
```
being used like:
```html
<my-element>
      This is child content
</my-element>
```
is
![image](https://user-images.githubusercontent.com/57605930/159590563-1acb3e61-2faf-45e8-8530-fae1bce33311.png)

and when I tried removing the shadow DOM with:
```ts
protected createRenderRoot(): Element | ShadowRoot {
    return this;
}
```
the end result is:
![image](https://user-images.githubusercontent.com/57605930/159590682-1f62c2fd-d9b5-4bb8-bf8a-c47e0a178eff.png)

Any ideas? I'm making a CSS theme, we cannot use `static styles` unfortunately

**Comment by lmtr0** [comment, marked as answer]
I have found that this makes the style work:
```js
protected createRenderRoot(): Element | ShadowRoot {
    let root = this.attachShadow({mode: 'open'})
    let link = document.createElement('link');
    link.setAttribute('rel', 'stylesheet');
    link.setAttribute('href', 'theme.css');
    root.appendChild(link);
    return root;
}

```

**Reply by cdelstad** [reply]
How is this for performance? Since it's needing to load the file, does it cause a delay in rendering the component?

**Comment by buehler** [comment]
If you use mor than one stylesheet or if you use webpack, which may rename the stylesheets:

I created a base class that essentially attaches all the stylesheets from the head. This way, even security headers are cloned as well.

```typescript
export class BaseElement extends LitElement {
  protected createRenderRoot(): Element | ShadowRoot {
    let root = this.attachShadow({ mode: 'open' });
    const links = document.head.getElementsByTagName('link');
    for (let i = 0; i < links.length; i++) {
      const link = links.item(i);
      if (!link || link.rel !== 'stylesheet') {
        continue;
      }
      root.appendChild(link.cloneNode(true));
    }
    return root;
  }
}
```

**Reply by buehler** [reply]
(works well with slot)

**Reply by lmtr0** [reply]
thx
